Assets/MixedRealityToolkit/_Core/DataProviders/SpatialObservers/Profiles/BaseMixedRealitySpatialObserverProfile.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Attributes;
+using Microsoft.MixedReality.Toolkit.Core.Definitions;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.DataProviders.SpatialObservers
+{
+    // No Scriptable Object Menu constructor attributes here, as this class is meant to be inherited.
+
+    /// <summary>
+    /// Base Mixed Reality Observer Profile.
+    /// </summary>
+    public abstract class BaseMixedRealitySpatialObserverProfile : BaseMixedRealityProfile
+    {
+        [SerializeField]
+        [Tooltip("How should the spatial awareness observer behave at startup?")]
+        private AutoStartBehavior startupBehavior = AutoStartBehavior.AutoStart;
+
+        /// <summary>
+        /// Indicates if the developer intends for the spatial awareness observer start immediately or wait for manual startup.
+        /// </summary>
+        public AutoStartBehavior StartupBehavior => startupBehavior;
+
+        [SerializeField]
+        [Tooltip("The dimensions of the spatial observer volume, in meters.")]
+        private Vector3 observationExtents = Vector3.one * 3;
+
+        /// <summary>
+        /// The size of the volume, in meters per axis, from which individual observations will be made.
+        /// </summary>
+        public Vector3 ObservationExtents => observationExtents;
+
+        [SerializeField]
+        [Tooltip("Should the spatial observer remain in a fixed location?")]
+        private bool isStationaryObserver = false;
+
+        /// <summary>
+        /// The size of the volume, in meters per axis, from which individual observations will be made.
+        /// </summary>
+        public bool IsStationaryObserver => isStationaryObserver;
+
+        [SerializeField]
+        [Tooltip("How often, in seconds, should the spatial observer update?")]
+        private float updateInterval = 3.5f;
+
+        /// <summary>
+        /// The frequency, in seconds, at which the spatial observer updates.
+        /// </summary>
+        public float UpdateInterval => updateInterval;
+
+        [PhysicsLayer]
+        [SerializeField]
+        [Tooltip("Which physics layer should this spatial observer use to create it's objects?")]
+        private int physicsLayer = -1;
```

Comment:
The default layer is whichever layer is availible for us to set the spatial observer layers on.

If the application is already using this layer we'll have problems.

The system is smart enough to setup the observer physics layers to the first availible layer and then update the default profile to it.

Reply:
> The system is smart enough to setup the observer physics layers to the first availible layer and then update the default profile to it.

Wouldnt that also work if the layer were defaulted to 31 (consistency with HTK)?

Reply:
There are some very good components and behaviors to be found in the HTK that customers have been successful with. Building upon these is important.

Reply:
And a benefit for both contributors and customers as time tested code and practices are reused.

Reply:
Yes, this is true, but I think something trivial as a layer assignment should be considered, esp when that can be changed. It needs to be configured by the user, and defaults set if none exist.

There's lots of pain points from HTK that we're trying to avoid as well. This is one of them.